In [1]:
from htmd import *

Please cite -- HTMD: High-Throughput Molecular Dynamics for Molecular Discovery
J. Chem. Theory Comput., 2016, 12 (4), pp 1845-1852. 
http://pubs.acs.org/doi/abs/10.1021/acs.jctc.6b00049


New HTMD version (1.1.0) is available. You are currently on (1.0.26). Use 'conda update htmd' to update to the new version.


In [2]:
mol = Molecule('3PTB')
mol.filter("protein")

2016-06-27 17:12:55,483 - htmd.molecule.molecule - INFO - Attempting PDB query for 3PTB
2016-06-27 17:12:56,421 - htmd.molecule.molecule - INFO - Removed 72 atoms. 1629 atoms remaining in the molecule.


KNOWN:
   - Donors:
        - ARG protonated* name NH1 NH2
        - LYS protonated* name NZ
   - Acceptors:
        - ASP deprotonated* name OD2 OD1
        - GLU deprotonated* name OE1 OE2

UNKNWON:
   - TYR name OH
   - HIS – There are 6 alternates for HIS, two rotomers and then 3 protonation states for each name ND1 NE2
   - CYS – disulfide bonded or not – name SG

CONFLICT:
   - ASN – There are 2 alternates – simple 180 degree rotations name ND2 OD1
   - GLN – There are 2 alternates – simple 180 degree rotations name OE1 NE2
   - SER name OG
   - THR name OG1

Backbone name O, N
   
UNCLASSIFIED: 
   - ALA
   - GLY
   - ILE
   - LEU
   - PRO 
   - VAL
   - MET

In [3]:
copied = mol.copy()

In [5]:
copied.filter('resname ARG LYS ASP GLU HIS CYS TYR')

2016-06-27 17:12:58,741 - htmd.molecule.molecule - INFO - Removed 1175 atoms. 454 atoms remaining in the molecule.


In [6]:
rid = np.unique(copied.get('resid')) # ia.resid also works!
rn = copied.get('resname', "name CA")

In [17]:
interest_selection = "name O N OH OG1 OG OE1 NE2 ND2 OE2 SG ND1 NH1 OD1 OD2 NZ NH2 and within 4 of name OH and resid "
for resname, resid in zip(rn, rid):
    if resname == "TYR":
        b = mol.get("resid", sel = interest_selection + str(resid))
        c = mol.get("resname", sel = interest_selection + str(resid))
        print(np.unique(b), np.unique(c), "resid", resid)
            

[ 20 159] ['LYS' 'TYR'] resid 20
[29] ['TYR'] resid 29
[37 39 60] ['LYS' 'SER' 'TYR'] resid 39
[59] ['TYR'] resid 59
[57 94] ['HIS' 'TYR'] resid 94
[151 192] ['GLN' 'TYR'] resid 151
[172 217 224 225] ['LYS' 'PRO' 'SER' 'TYR'] resid 172
[159 184] ['LYS' 'TYR'] resid 184
[190 228] ['SER' 'TYR'] resid 228
[101 179 234] ['ASN' 'TYR'] resid 234


TYR:
<br>
20 - LYS 159 NZ. No problem
<br>
29 No problem, do not interact
<br>
39 - LYS 60 NZ and SER 37 OG (SER no seems a probable interaction)
<br>
59 No problem, do not interact
<br>
94 - HIS 57 N
<br>
151 - GLN 192 O 
<br>
172 - SER 217 N and LYS 224 O and pro 225 O
<br>
184 - LYS 159 O
<br>
228 - SER 190 OG
<br>
234 - ASN 101 ND2 and ASN 179 ND2 (?)